### PR TITLE
Rename sig/federation to sig/multicluster

### PR DESCRIPTION
**Pull request by spiffxp**
Noticed this while filing kubernetes/kubernetes#57911

### CODE REVIEW

**Comment by spiffxp** [comment]
/hold
I'd like to make sure someone from SIG multicluster signs off on this as well

**Comment by cblecker** [comment]
/lgtm
^^ from a correctness perspective

File: label_sync/labels.yaml [hunk 1]
```diff
@@ -128,12 +128,15 @@ labels:
...
-  - color: d2b48c
-    name: sig/federation
   - color: d2b48c
     name: sig/gcp
   - color: d2b48c
     name: sig/instrumentation
+  - color: d2b48c
+    name: sig/multicluster
+    previously:
+      - name: sig/federation
```
Comment (nikhiljindal) [inline label_sync/labels.yaml]: Remove this? Not sure why is the name listed twice.

Comment (spiffxp) [reply label_sync/labels.yaml]: Looking at k/k _both_ of these labels show up right now, so this is intended to catch both
- https://github.com/kubernetes/kubernetes/labels/sig%2Ffederation%20%28deprecated%20-%20do%20not%20use%29
- https://github.com/kubernetes/kubernetes/labels/sig%2Ffederation

We can remove one for the other later

**Comment by nikhiljindal** [comment]
cc @csbell @quinton-hoole

**Comment by spiffxp** [comment]
@fejta I'm confused, I still see https://github.com/kubernetes/kubernetes/labels/sig%2Ffederation%20%28deprecated%20-%20do%20not%20use%29 applied to 800+ closed issues, does label_sync only work against open issues? should I expect to manually delete migrated labels myself once the job has run?

**Pull request merged**